### DISCUSSION TITLE
Best practice for calling vehicle_data endpoint

### DISCUSSION THREAD

**Discussion by Urkman** [timdorr/tesla-api#721] [Q&A]
What is you practice for calling the vehicle_data endpoint to always have the latest vehicle data?
Does calling this endpoint prevents the Tesla from going to sleep?
For me it looks like, when calling this endpoint every minute the Tesla is going to sleep some minutes later...

So, how do you handle this?

**Comment by GaPhi** [comment]
My current implementation :
- Use "let_sleep=true" query argument and specify only the required endpoints to the query...
- Polling at 10s period when driving
- Polling at 10s when running my own cabin overheat protection mode
- Polling immediately on specific requests (after any other command that could modify the vehicle state)
- Polling at 60s period otherwise
- Never query more than once by 5s period : I delay a single request in that case at 5s and discard any new polling request but merge the endpoints to refresh during this period

**Reply by Urkman** [reply]
Thanks for your reply :)
I didn't know about the "let_sleep" query argument. Where do you know it from?

So the url should look like this: "GET https://owner-api.teslamotors.com/api/1/vehicles/{vehicle_id}/vehicle_data?let_sleep=true"
Right?

**Reply by superfloh247** [reply]
does this also work with the FleetAPI?
Looks like my Model S 85 MCU1 does not fall asleep while I poll this endpoint

**Comment by GaPhi** [comment]
Yes that's it.
To find arguments of any request, the easy way is just to have a look at
the log files produced by the tesla application (you have to jailbreak your
android smartphone to access the directory)
Another solution is to reverse-engineer the apk file and decompile java
code (also possible with native code)... it is more accurate to understand
the various special algorithm cases to handle.

Le sam. 13 mai 2023 à 16:27, Stefan Sturm ***@***.***> a
écrit :
